src/algos/time_iteration_direct.jl

```diff
@@ -68,45 +74,49 @@ function time_iteration_direct(model, dprocess::AbstractDiscretizedProcess,
     ###############################   Iteration loop
     n_h = length(model.symbols[:expectations])
 
+    # temporary vars
+    E_f = DRStore(Point{n_h}, [length(e) for e in ds0.data])
+    d_E_f = deepcopy(E_f.data[1])
+    S = deepcopy(s)
+    X = deepcopy(ds0.data[1])
+
 
     while it<maxit && err>tol_η
 
         it += 1
 
         tic()
 
-        E_f = [zeros(Point{n_h},N) for i=1:number_of_smooth_drs(dprocess)]
+        set_values!(dr, [ds0.data...])
```

Comment:
Is there any kind of penalty from the `[ds0.data...]` that appears throughout? (e.g. performance, memory, or type inference penalty)

This might be pushing things too far (and please tell me if it is), but would it make sense to have `drs.data` be `NTuple{N,Vector{T}}` so we know how many elements are in it and can use `@generated` functions to unroll these `[drs.data...]` calls?

Reply:
For now I don't think we have any penalty. The elements of `[ds0.data...]` should be correctly inferred. `set_values!` itself  could be improved to take a tuple as argument. I doubt unrolling would provide much benefit, as the amount of code within the loop is quite big.